espwrap/adaptors/sendgrid_v3.py

```diff
@@ -209,7 +210,7 @@ def generate_subject(self, email, name):
         try:
             username = email.split('@')[0]
             if '.' in username or '+' in username:
-                return Subject('{} [{}]'.format(self.subject.decode('utf-8'), email.decode('utf-8')))
+                return Subject(u'{} [{}]'.format(self.subject, email))
```

Comment:
the `u` is unnecessary due to `unicode_literals` import above

Reply:
think it depends on what the caller is sending and how the caller is encoding the original string... in the following scenario, w/o the u'' it will throw the original error. There was an error we fixed prior to this that was using some other thing than u''... guess this is the best failsafe not knowing how folks are sending to this function

```
# -*- coding: utf-8 -*-
name = u'Pasquale’s Pizza and Pasta'
print(name)
string = u'Merchant PPM Boarding Errors: {}'.format(name)

print(string)
x = '{}'.format(string) # without u''
print(string)
```

>>>
Pasquale’s Pizza and Pasta
Merchant PPM Boarding Errors: Pasquale’s Pizza and Pasta
Traceback (most recent call last):
  File "test.py", line 2332, in <module>
    x = '{}'.format(string)
UnicodeEncodeError: 'ascii' codec can't encode character u'\u2019' in position 38: ordinal not in range(128)

```
# -*- coding: utf-8 -*-
name = u'Pasquale’s Pizza and Pasta'
print(name)
string = u'Merchant PPM Boarding Errors: {}'.format(name)

print(string)
x = u'{}'.format(string)  # with u''
print(string)
```
>>>
Pasquale’s Pizza and Pasta
Merchant PPM Boarding Errors: Pasquale’s Pizza and Pasta
Merchant PPM Boarding Errors: Pasquale’s Pizza and Pasta

Reply:
open to suggestions...

Reply:
@mwspoton does your test code use `from __future__ import unicode_literals` at the top of the import list?

Reply:
hrm... that would fix it... can remove the u' and hopefully we would only need to update the caller functions going forward

Reply:
works on my end...